File: Asklepios/Asklepios.Core/Entities/Patients/Patient.cs
using Asklepios.Core.Entities.Departments;
using Asklepios.Core.Entities.Examinations;

namespace Asklepios.Core.Entities.Patients;

public class Patient
{
    public Guid PatientId { get; set; }
    public string PatientName { get; set; }
    public string PatientSurname { get; set; }
    public string PeselNumber { get; set; }
    public string InitialDiagnosis { get; set; }
    public bool IsDischarged { get; set; }
    public string Treatment { get; set; }
    
    public Guid DepartmentId { get; set; }
    public Department Department { get; set; }
    
    public Guid RoomId { get; set; }
    public Room Room { get; set; }
    
    public IEnumerable<ExamResult> ExamResults { get; set; }
    
    public IEnumerable<Operation> Operations { get; set; }

    public Patient()
    {
    }

    public Patient(Guid patientId, string patientName, string patientSurname, string peselNumber, string initialDiagnosis, bool isDischarged, string treatment, Guid departmentId, Guid roomId)
    {
        PatientId = patientId;
        PatientName = patientName;
        PatientSurname = patientSurname;
        PeselNumber = peselNumber;
        InitialDiagnosis = initialDiagnosis;
        IsDischarged = isDischarged;
        Treatment = treatment;
        DepartmentId = departmentId;
        RoomId = roomId;
    }
}

File: Asklepios.Web/src/models/Patients/patient.ts
import {OperationDto} from "@/models/Examinations/operation"
import { ExamResultDto } from "../Examinations/examResult";

export interface PatientDto {
  patientId: string;
  patientName: string;
  patientSurname: string;
  peselNumber: string;
  initialDiagnosis: string;
  isDischarged: boolean;
  treatment: string;
  departmentId: string;
  roomId: string;
  admissionDate: Date;
  address: string;
}

export interface PatientListDto {
  patientId: string;
  patientName: string;
  patientSurname: string;
  peselNumber: string;
  initialDiagnosis: string;
  isDischarged: boolean;
  treatment: string;
  departmentName: string;
  roomNumber: string;
}

export interface PatientDetailsDto extends PatientDto {
  operations: OperationDto[];
  examResults: ExamResultDto[];
}

export interface PatientAutocompleteDto {
  patientId: string;
  patientName: string;
  patientSurname: number;
  peselNumber: string;
}

export class InputCreatePatient {
  patientName: string = '';
  patientSurname: string = '';
  peselNumber: string = '';
  initialDiagnosis: string = '';
  isDischarged: boolean = false;
  treatment: string = '';
  departmentId: string = '';
  roomId: string = '';
  medicalStaffId: string = '';
  address: string = '';
}


